Code/Framework/AzCore/CMakeLists.txt

```
@@ -20,6 +20,25 @@ if(PAL_TRAIT_PROF_PIX_SUPPORTED)
     endif()
 endif()
 
+# By default, the Settings Registry will apply user and project registry files as overrides in development (non-release) builds,
+# and will ignore all overrides in release builds.
+# This behavior can be overridden by passing -DALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES=0, 1, 2 or 3 when generating the build files.
+# The override can be removed by passing -UALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES when generating the build files.
+set(ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES "" CACHE STRING 
+"Forces the Settings Registry development overrides to be used or ignored.
+  If unset, development overrides are used in development builds and ignored in release builds.
+  0 disables the development overrides in all builds.
+  1 = registry overrides are allowed from the command line.
+  2 = registry overrides are allowed from the command line, engine, gem, and project files.
+  3 = registry overrides are allowed from the command line, engine, gem, project, and user files.")
+
+# If there's a value in ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES, the compiler flag will get set to it.
+# Otherwise, the variable will be empty and no compiler flag will be set, leaving it to the code to decide
+# whether or not to allow the Settings Registry development overrides.
+set(ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES_FLAG $<$<NOT:$<STREQUAL:"${ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES}","">>:ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES=$<BOOL:${ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES}>>)
```

In this case we don't want to convert the Cache variable value into a boolean here, so we should remove the `$<BOOL: ` generator expression
```suggestion
set(ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES_FLAG $<$<NOT:$<STREQUAL:"${ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES}","">>:ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES=${ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES}>)
```